wpimath/src/main/native/cpp/geometry/CoordinateAxis.cpp

```diff
@@ -11,25 +11,31 @@ CoordinateAxis::CoordinateAxis(double x, double y, double z) : m_axis{x, y, z} {
 }
 
 CoordinateAxis CoordinateAxis::N() {
-  return CoordinateAxis{1.0, 0.0, 0.0};
+  static CoordinateAxis instance{1.0, 0.0, 0.0};
+  return instance;
```

Comment:
This just returns a copy, so it's going to construct a copy anyway, right?  Or are Eigen vectors lazily copied?

Reply:
It's a deep copy, but it avoids the normalization in the CoordinateAxis constructor. For CoordinateSystem, it avoids a bunch of math for converting the basis vectors into a Rotation3d (a unit quaternion).

I considered returning by const ref. Should it?

Reply:
The static should also be made const.